### PR TITLE
Rework the sections and subsections of our docs

### PR DESCRIPTION
**Pull request by marga-kinvolk**
# Rework the sections and subsections of our docs

This is a major change on how the docs were organized.  The existing structure had been raised by a couple of clients as confusing (mostly because how much emphasis was being put on "clusters"), and I was personally having a very hard time navigating it.

So, I've re-distributed the different pages into sections that made more sense to me and that hopefully will also make more sense to clients.  The main sections are now:
 * Installing (includes the "quickstart", and subsections for Cloud Providers, Virtual Machines and Bare Metal)
 * Provisioning Tools (Ignition, Terraform, Cloud Config, etc)
 * Setup and Operations (includes common customizations, managing clusters, managing storage, etc).
 * Container Runtimes
 * Migrating from CoreOS
 * Reference.

In all moved pages, I've included aliases to the previous location, so that any links already pointing to the previous paths should still work.

## Included changes

On top of changing the structure of the site, I also:
1. Removed the tables from the main index page. These tables didn't go well with the new format.  Instead, the index file is now mostly a plain list of links.
2. Fixed links throughout (most of them had been broken with the migration to the new docs).
3. Updated all github repos to point to kinvolk rather than flatcar-linux
4. Added descriptions to the metadata of most pages so that we can use the nice automatically generated indexes provided by Hugo (the Overview pages now include a list of the pages inside with the descriptions provided).
5. Fixed random errors, typos, outdated content when I came across them.

## Why this is so big

I know this is a large refactor, I'm sorry. Splitting it into pieces is hard because files were moved from many different places, and so moving files around requires the new sections to exist; fixing links requires the files to be in their definitive place (otherwise they need to be fixed several times); and in general I went through all files checking links which meant I also did small fixes here and there and the effort to pull them out was really not worth it.

# How to use / Testing done

I've been running a local version of the website, checking that the rendered structure makes sense as well as testing that links work correctly.

### CODE REVIEW

**Review by sayanchowdhury — CHANGES_REQUESTED** [review]
💯 on this PR.

A couple of comments. I commented on a couple of kinvolk/etcd references but also skipped a few. Since kinvolk/etcd is not regularly updated it would be better to point to upstream docs.

File: docs/setup/systemd/udev-rules.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
 ---
 title: Using systemd and udev rules
-weight: 10
+description: How to run untis when specific udev events trigger.
```
[the remark this answers is not in the document]
Comment (marga-kinvolk) [reply docs/setup/systemd/udev-rules.md]: Done, thanks for catching this.

File: docs/setup/security/generate-self-signed-certificates.md [hunk 2]
```diff
@@ -291,10 +293,10 @@ openssl x509 -in client.pem -text -noout
 
 ## More information
 
-For another examples, check out these documents:
+For more examples, check out these documents:
 
-[Custom Certificate Authorities](adding-certificate-authorities)
-[etcd Security Model](https://github.com/flatcar-linux/etcd/blob/master/Documentation/op-guide/security.md)
+ * [Custom Certificate Authorities](adding-certificate-authorities)
+ * [etcd Security Model](https://github.com/kinvolk/etcd/blob/master/Documentation/op-guide/security.md)
```
[the remark this answers is not in the document]
Comment (marga-kinvolk) [reply docs/setup/security/generate-self-signed-certificates.md]: Sure, makes a lot of sense, thanks!

File: docs/setup/clusters/discovery.md [hunk 3]
```diff
@@ -136,8 +142,8 @@ The public discovery service is just an etcd cluster made available to the publi
...
 
-If you're interested in how discovery API works behind the scenes in etcd, read about [etcd clustering](https://github.com/flatcar-linux/etcd/blob/master/Documentation/op-guide/clustering.md).
+If you're interested in how discovery API works behind the scenes in etcd, read about [etcd clustering](https://github.com/kinvolk/etcd/blob/master/Documentation/op-guide/clustering.md).
 
-[etcd-reconf]: https://github.com/flatcar-linux/etcd/blob/master/Documentation/op-guide/runtime-configuration.md
-[etcd-reconf-no-disc]: https://github.com/flatcar-linux/etcd/blob/master/Documentation/op-guide/runtime-reconf-design.md#do-not-use-public-discovery-service-for-runtime-reconfiguration
+[etcd-reconf]: https://github.com/kinvolk/etcd/blob/master/Documentation/op-guide/runtime-configuration.md
+[etcd-reconf-no-disc]: https://github.com/kinvolk/etcd/blob/master/Documentation/op-guide/runtime-reconf-design.md#do-not-use-public-discovery-service-for-runtime-reconfiguration
 [etcd-reconf-on-flatcar]: https://github.com/coreos/docs/blob/master/etcd/etcd-live-cluster-reconfiguration.md
```
[the remark this answers is not in the document]
Comment (marga-kinvolk) [reply docs/setup/clusters/discovery.md:149]: Agreed. Filed https://github.com/kinvolk/Flatcar/issues/306 to track that.

File: docs/setup/security/adding-certificate-authorities.md [hunk 4]
```diff
@@ -20,4 +22,4 @@ The setup process for any of these use-cases is the same:
 
 [Generate Self-Signed Certificates](generate-self-signed-certificates)
 
-[etcd Security Model](https://github.com/flatcar-linux/etcd/blob/master/Documentation/op-guide/security.md)
+[etcd Security Model](https://github.com/kinvolk/etcd/blob/master/Documentation/op-guide/security.md)
```
[the remark this answers is not in the document]
Comment (marga-kinvolk) [reply docs/setup/security/adding-certificate-authorities.md]: Done, thanks.

File: docs/setup/security/hardening-guide.md [hunk 5]
```diff
@@ -55,7 +58,7 @@ SELinux is a fine-grained access control mechanism integrated into Flatcar Conta
 
 Flatcar Container Linux implements SELinux, but currently does not enforce SELinux protections by default. The [SELinux on Flatcar Container Linux guide][selinux-guide] covers the process of checking containers for SELinux policy compatibility and switching SELinux into enforcing mode.
 
-[smt-guide]: clusters/securing/disabling-smt
-[sshd-guide]: clusters/securing/customizing-sshd
-[etcd-sec-guide]: https://github.com/flatcar-linux/etcd/blob/v3.2.11/Documentation/op-guide/security.md
-[selinux-guide]: clusters/securing/selinux
+[smt-guide]: disabling-smt
+[sshd-guide]: customizing-sshd
+[etcd-sec-guide]: https://github.com/kinvolk/etcd/blob/v3.2.11/Documentation/op-guide/security.md
```
[the remark this answers is not in the document]
Comment (marga-kinvolk) [reply docs/setup/security/hardening-guide.md]: Done, thanks.

**Comment by marga-kinvolk** [comment]
After fixing the links to point to github.com/etcd-io/etcd, I realized that there's actually a docs site for etcd docs (etcd.io/docs). It makes a lot more sense to point there rather than the github sources, so fixed that now.

**Review by sayanchowdhury — APPROVED** [review]
LGTM

**Comment by marga-kinvolk** [comment]
Vincent/Joaquim: I intend to submit this tomorrow noon-ish. Please let me know if you intend to review and want me to wait some more.  We can always fix things later, but as this is a large rework, I'm ok waiting a bit more to make sure we're in agreement regarding the new structure.

Cheers.

**Comment by vbatts** [comment]
right on. How are we testing this repo now? do i pull the kinvolk/website and `make run` or ...?

**Comment by marga-kinvolk** [comment]
You need both the website and this repo. From the website repo, you need to do `make docs`, `make run` and use the `tools/preview_docs.sh` script.  I'm not sure in which order, I think I just ran all those commands repeatedly until it worked :).

The documentation on how to do it is here: https://github.com/kinvolk/website#testing-documentation-locally

**Comment by vbatts** [comment]
oh this is much nicer
LGTM

**Comment by joaquimrocha** [comment]
> You need both the website and this repo. From the website repo, you need to do `make docs`, `make run` and use the `tools/preview_docs.sh` script. I'm not sure in which order, I think I just ran all those commands repeatedly until it worked :).
> 
> The documentation on how to do it is here: https://github.com/kinvolk/website#testing-documentation-locally

It's: `make docs`, then the `tools/preview_docs.sh`, then `make run`.

FWIW this is what I expected the OS & Sec to look into last summer (just to clarify that the current structure so far has been a rough proposal). The changes LGTM.

**Pull request merged**